tests/sequelize_test.js

```diff
@@ -0,0 +1,146 @@
+'use strict';
+
+const { expect, assert } = require('chai');
+const { DataTypes, Sequelize } = require('../source');
+const dialect = 'postgres';
+const _ = require('lodash');
+
+const qq = str => {
+  if (dialect === 'postgres' || dialect === 'mssql') {
+    return `"${str}"`;
+  }
+  if (dialect === 'mysql' || dialect === 'mariadb' || dialect === 'sqlite') {
+    return `\`${str}\``;
+  }
+  return str;
+};
+
+const config = {
+  rand: () => {
+    return parseInt(Math.random() * 999, 10);
+  }
+};
+
+describe('Sequelize', () => {
+  describe('truncate', () => {
+    // Edit reason:
+    // CRDB does not guarantee ids to start at 1
+    it('truncates all models', async function () {
+      const Project = this.sequelize.define(`project${config.rand()}`, {
+        id: {
+          type: DataTypes.INTEGER,
+          primaryKey: true,
+          autoIncrement: true
+        },
+        title: DataTypes.STRING
+      });
+
+      await this.sequelize.sync({ force: true });
+      // Added ID, as it is expected to be 1
+      const project = await Project.create({ id: 1, title: 'bla' });
+      expect(project).to.exist;
+      expect(project.title).to.equal('bla');
+      expect(project.id).to.equal(1);
+      await this.sequelize.truncate();
+      const projects = await Project.findAll({});
+      expect(projects).to.exist;
+      expect(projects).to.have.length(0);
+    });
+  });
+
+  describe('sync', () => {
+    // Edit reason:
+    // Error message CRDB provides does not quote "bar"
+    it('fails with incorrect database credentials (1)', async function () {
+      this.sequelizeWithInvalidCredentials = new Sequelize(
+        'omg',
+        'bar',
+        null,
+        _.omit(this.sequelize.options, ['host'])
+      );
+
+      const User2 = this.sequelizeWithInvalidCredentials.define('User', {
+        name: DataTypes.STRING,
+        bio: DataTypes.TEXT
+      });
+
+      try {
+        await User2.sync();
+        expect.fail();
+      } catch (err) {
+        assert(
+          [
+            'fe_sendauth: no password supplied',
+            'role "bar" does not exist',
+            'FATAL:  role "bar" does not exist',
+            'password authentication failed for user "bar"',
+            // Added the error message generated by CRDB
+            'password authentication failed for user bar'
+          ].includes(err.message.trim())
+        );
+      }
+    });
+  });
+
+  describe('define', () => {
+    describe('transaction', () => {
+      beforeEach(async function () {
+        const sequelize = await this.sequelize;
+        this.sequelizeWithTransaction = sequelize;
+      });
+
+      // Skip reason:
+      // CRDB allows only one transaction per node.
```

Comment:
this comment isn't true, I think we should change this to CRDB transactions have stronger locking restrictions so that a transaction writing will block reads on the same object from a different ransaction.